mesop/labs/text_to_image.py

```diff
@@ -0,0 +1,132 @@
+import types
+from typing import Callable, Generator, cast
+
+import mesop as me
+
+
+@me.stateclass
+class State:
+  input: str
+  output: str
+  textarea_key: int
+
+
+def text_to_image(
+  transform: Callable[[str], Generator[str, None, None] | str],
+  *,
+  title: str | None = None,
+):
+  """Creates a simple UI which takes in a text input and returns an image output.
+
+  This function creates event handlers for text input and output operations
+  using the provided function `transform` to process the input and generate the image
+  output.
+
+  Args:
+    transform: Function that takes in a string input and returns a URL to an image
+    or a base64 encoded image.
+    title: Headline text to display at the top of the UI.
+  """
+
+  def on_input(e: me.InputEvent):
+    state = me.state(State)
+    state.input = e.value
+
+  def on_click_generate(e: me.ClickEvent):
+    state = me.state(State)
+    output = transform(state.input)
+    if isinstance(output, types.GeneratorType):
```

Comment:
For image, it's a little less obvious the use case for generators. Typically, I'd expect the image to be returned all in one piece?

WDYT about having output being a string first, and then making it support generators if we see a use case?

Reply:
Good catch. I agree that accepting string only for now is good.